### PR TITLE
.NET Core port - DO NOT MERGE

### PR DESCRIPTION
**Pull request by darkl**
Hi,

I tried for fun to make this library be .NET Core compatible.

I made a project.json file and made it compile, note that I changed some things: there is no such thing "InvariantCultureIgnoreCase", so I replaced it with "OrdinalIgnoreCase",
there are no more "Close" methods for streams and sockets, so I replaced it with "Dispose".

I also didn't reimplement QueuedStream since I'm not sure if this hack is needed for .NET Core. I also didn't covert the tests.

I didn't try this, I don't know if it works.

I'm opening this pull request, just for the case that anyone wants to continue this work on the port. Please don't merge it until tests are converted, pass, and until this is actually tested.

Best regards,
Elad

### CODE REVIEW

**Comment by statianzo** [comment]
Slick work.

**Comment by AdrianBathurst** [comment]
Thanks!! I'll be needing this port soon, so thank you very much!

**Comment by NVentimiglia** [comment]
+1

DotNetCore/Kestral version of my server dropped to 5k RPS... from an easy 50k.

**Comment by NVentimiglia** [comment]
The dependency NETStandard.Library >= 1.6.0 could not be resolved.	Fleck	G:\Drive\Avarice\SignalMQ\Fleck-master\core\Fleck\project.json	5

**Comment by NVentimiglia** [comment]
After some fiddling, I got the project down to 4 errors with the following

`
{
  "title": "Fleck",
  "version": "1.0.0-*",

  "dependencies": {
    "NETStandard.Library": "1.6.0"
  },

  "frameworks": {
    "netstandard1.6": {
      "imports": "dnxcore50"
    }
  },

  "buildOptions": {
    "allowUnsafe": true,    "define": [ "NETCORE" ]
  }
}
`

The errors being the removal of the BeginAsync/EndAsync and missing System.Security.SSLStream

Do we have an ETA when this will be ready for testing. I recently got the microsoft websocket up to 800,000 OPS would like to test out fleck again. @darkl

**Comment by darkl** [comment]
@NVentimiglia, I worked on this a bit today.

This works fine on my machine with .NET RTM:

>>> dotnet restore
>>> dotnet build src\Fleck src\Samples\ConsoleApp
>>> cd src\Samples\ConsoleApp
>>> dotnet run

Tests don't compile since a .NET Core compatible Moq isn't available yet.

I'm not sure if I'll finish this, but you're welcome to continue from where I've stopped.

Elad

**Comment by NVentimiglia** [comment]
7/9/2016 12:31:43 PM [Info] Server started at ws://0.0.0.0:5000 (actual port 5000)
One or more errors occurred. (Attempt by method 'Fleck.SocketWrapper.Accept(System.Action`1<Fleck.ISocket>, System.Action`1<System.Exception>)' to access method 'System.Net.Sockets.Socket.BeginAccept(System.AsyncCallback, System.Object)' failed.)
Terminating

**Comment by darkl** [comment]
Are you using the latest version? (netcore branch)
The latest version doesn't call BeginAccept at all.

Elad

File: src/Fleck/QueuedStream.cs [hunk 1]
```diff
@@ -2,9 +2,127 @@
...
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            _stream.Write(buffer, offset, count);
+        }
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return _stream.WriteAsync(buffer, offset, count, cancellationToken);
```
Comment (bilal-fazlani) [inline src/Fleck/QueuedStream.cs]: Shouldn't this be awaited ?

File: src/Fleck/QueuedStream.cs [hunk 2]
```diff
@@ -2,9 +2,127 @@
...
+        }
+
+        public override void Flush()
+        {
+            _stream.Flush();
+        }
+
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return _stream.FlushAsync(cancellationToken);
```
Comment (bilal-fazlani) [inline src/Fleck/QueuedStream.cs]: Shouldn't this be awaited ?

File: src/Fleck/QueuedStream.cs [hunk 3]
```diff
@@ -2,9 +2,127 @@
...
+        readonly Stream _stream;
+
+        public QueuedStream(Stream stream)
+        {
+            _stream = stream;
+        }
+
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            return _stream.CopyToAsync(destination, bufferSize, cancellationToken);
```
Comment (bilal-fazlani) [inline src/Fleck/QueuedStream.cs]: Shouldn't this be awaited ?

Comment (statianzo) [inline src/Fleck/QueuedStream.cs]: awaiting would just wrap the CopyToAsync task with another task that does nothing but return the original result. Why do you think it should be wrapped?

Comment (bilal-fazlani) [inline src/Fleck/QueuedStream.cs]: Never really thought of it that way. You are right.

**Comment by darkl** [comment]
Not needed anymore.

**Pull request closed without merging**